cmd/desync/list.go

```diff
@@ -41,15 +50,56 @@ func runList(ctx context.Context, opt listOptions, args []string) error {
 	if err != nil {
 		return err
 	}
-	// Write the list of chunk IDs to STDOUT
+
+	var chunksInfo []desync.ChunkAdditionalInfo
+
+	var s desync.LocalStore
+	if opt.store != "" {
+		options, err := cfg.GetStoreOptionsFor(opt.store)
+		if err != nil {
+			return err
+		}
+		// Use compressed chunks, if available, to record their sizes
+		options.Uncompressed = false
+		s, err = desync.NewLocalStore(opt.store, options)
+		if err != nil {
+			return err
+		}
+	}
+
 	for _, chunk := range c.Chunks {
-		fmt.Fprintln(stdout, chunk.ID)
+		var size int64 = 0
+		if opt.detailed && opt.store != "" {
+			size, err = s.GetChunkSize(chunk.ID)
+			if err != nil {
+				// The compressed chunk is either missing or unreadable from the provided local store
+				size = 0
+			}
+		}
+
+		chunksInfo = append(chunksInfo, desync.ChunkAdditionalInfo{
+			ID:               chunk.ID.String(),
```

Comment:
Let's try to avoid using `.String()` so frequently. If you change the type of `ID` in `ChunkAdditionalInfo` to `ChunkID` this will integrate into the rest of the code base a lot better. You may need to add [custom Text marshaller/unmarshaller](https://pkg.go.dev/encoding/json#example-package-TextMarshalJSON) to the ChunkID type to have the ID as string in the JSON file.

Reply:
It should be done now.